### PR TITLE
Make bridge and zome calls consistent

### PR DESCRIPTION
**Pull request by willemolding**
## PR summary

This addresses an issue we have been having with bridge calls in developing Personas/Profiles.

Prior to this if calling a function with a signature `fn_name(pamams) -> ZomeApiResult<SomeType>` using `hdk::call` from another zome in the same DNA the result would be a JsonString-ified `ZomeApiResult<SomeType>`. If calling the same function over a bridge with `hdk::call` the result would be a JsonString-ified `SomeType`.

I understand that in the past it was necessary to do this unwrapping in code as it was difficult to do in the zome but the conversion is now trivial with #1464 .

In the current implementation not only is it inconsistent but it does not account for the case where a zome call may not return a ZomeApiResult (such as functions which cannot fail)

This PR removes the unwrapping in the core and updates the app spec to reflect the change. The updated app spec now shows consistency between zome and bridge calls.

@lucksus in particular I would like to discuss this with you

## changelog

Please check one of the following, relating to the [CHANGELOG-UNRELEASED.md](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md)

- [x] this is a code change that effects some consumer (e.g. zome developers) of holochain core so it is added to the CHANGELOG-UNRELEASED.md (linked above), with the format `- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)`
- [ ] this is not a code change, or doesn't effect anyone outside holochain core development

### CODE REVIEW

File: core/src/nucleus/ribosome/api/call.rs [hunk 1]
```diff
@@ -139,32 +135,8 @@ fn bridge_call(runtime: &mut Runtime, input: ZomeFnCallArgs) -> Result<JsonStrin
...
-                    // one last time:
-                    let maybe_parsed_value: Result<Value, serde_json::error::Error> =
-                        serde_json::from_str(&sanitized_response);
-                    match maybe_parsed_value {
-                        Ok(parsed_value) => Ok(JsonString::from(parsed_value)),
-                        Err(_) => Ok(JsonString::from(RawString::from(sanitized_response))),
-                    }
-                })
+            // This should be returned as a JsonString for handling in the zome code.
+            Ok(JsonString::from_json(&sanitized_response))
```
Comment (willemolding) [inline core/src/nucleus/ribosome/api/call.rs]: This is the main change to core. The unwrapping code is dropped and the JsonString result returned directly as for a zome call

**Comment by philipbeadle** [comment]
How does this affect the Holo apps that bridge? I agree we should be consistent across the calls.

**Comment by philipbeadle** [comment]
This works so well!! The code in basic-chat to call P&P across the bridge and across zomes is super simple and readable now.

**Comment by lucksus** [comment]
I'm very glad if we can do this unwrapping/deserializing in WASM now - and even that easy!
I had added the unwrapping code by responding to a pull from Holo and Julio in particular. With that I was initially making bridge and zome calls consistent, but, yes, only for the Results known by that unwrapping code.. ;)

I would like to ping people in Holo who are using bridges right now if they agree this is better. They will have to make the same changes using `.try_into()` as you did here in app-spec.

But I'm all for this change!

@peeech, @pjkundert, @artbrock (who else?)

**Comment by zippy** [comment]
whoa, something broke in the last last merge develop

**Pull request merged**